src/Config.php

```diff
@@ -148,7 +174,668 @@ public static function concurrencySet(int $value): void
     public static function version(): string
     {
-        return vips_version();
+        Config::ffi();
+
+        return self::$library_major . "." .
+            self::$library_minor . ".".
+            self::$library_micro;
+    }
+
+    /**
+     * Handy for debugging.
+     */
+    public static function printAll()
+    {
+        Config::ffi()->vips_object_print_all();
+    }
+
+    public static function ffi()
+    {
+        if (!self::$ffi_inited) {
+            self::init();
+            self::$ffi_inited = true;
+        }
+
+        return self::$ffi;
+    }
+
+    public static function ctypes(string $name)
+    {
+        Config::ffi();
+
+        return self::$ctypes[$name];
+    }
+
+    public static function gtypes(string $name)
+    {
+        Config::ffi();
+
+        return self::$gtypes[$name];
+    }
+
+    public static function ftypes(string $name)
+    {
+        Config::ffi();
+
+        return self::$ftypes[$name];
+    }
+
+    /**
+     * Throw a vips error as an exception.
+     *
+     * @throws Exception
+     *
+     * @return void
+     *
+     * @internal
+     */
+    public static function error(string $message = "")
+    {
+        if ($message == "") {
+            $message = Config::ffi()->vips_error_buffer();
+            Config::ffi()->vips_error_clear();
+        }
+        $exception = new Exception($message);
+        Utils::errorLog($message, $exception);
+        throw $exception;
+    }
+
+    /**
+     * Shut down libvips. Call this just before process exit.
+     *
+     * @throws Exception
+     *
+     * @return void
+     *
+     * @internal
+     */
+    public static function shutDown()
+    {
+        Config::ffi()->vips_shutdown();
+    }
+
+    public static function filenameGetFilename($name)
+    {
+        $pointer = Config::ffi()->vips_filename_get_filename($name);
+        $filename = \FFI::string($pointer);
+        Config::ffi()->g_free($pointer);
+
+        return $filename;
+    }
+
+    public static function filenameGetOptions($name)
+    {
+        $pointer = Config::ffi()->vips_filename_get_options($name);
+        $options = \FFI::string($pointer);
+        Config::ffi()->g_free($pointer);
+
+        return $options;
+    }
+
+    public static function atLeast($need_major, $need_minor)
+    {
+        return $need_major < self::$library_major ||
+            ($need_major == self::$library_major &&
+             $need_minor <= self::$library_minor);
+    }
+
+    private static function init()
+    {
+        $library_name = "libvips";
+
+        switch (PHP_OS_FAMILY) {
+            case "Windows":
+                $library_ext = ".dll";
+                break;
+
+            case "OSX":
+                $library_ext = ".dylib";
+                break;
+
+            default:
+                $library_ext = ".so";
+                break;
+        }
+
+        $vipshome = getenv("VIPSHOME");
+        if ($vipshome) {
+            $library_location = $vipshome . "/lib/";
```

Comment:
Perhaps we should rely always on FFI's search? AFAIK, the other bindings doesn't have this logic.

(I still had the `export VIPSHOME=/usr` in `~/.bashrc`, which caused test failures because Fedora expects this to point to `/lib64` instead)

Reply:
You're probably right, it's just confusing.

I suppose we'll need to search for "libvips42.dll" on windows, I'll try to add that.